doc/source/guide/in_depth/functional_guide.rst

```diff
@@ -0,0 +1,257 @@
+.. _functional_in_depth:
+
+#####################
+Functional
+#####################
+
+A *functional* is an operator ``f`` that maps from some vector space ``X`` to the field of scalars ``F`` associated with the vector space:
+
+.. math::
+   
+   f : X \to F.
+
+In the ODL solver package, functionals are implemented in the `Functional` class, a subclass to `Operator`.
+
+From a mathematical presepective, the above is a valide definition of a functional.
+However, since the purpose of these functionals are primarely to be used for solving optimization problems, the following assumptions are made:
+
+ * the vector space ``X`` is a Hilbert space.
```

Comment:
Not at all sure if I like this being built into odl this strongly. Why do we need to assume this? Is it not better to check this in each of the functionals individually? If we build this in now it will be hard to change later.